lua/obsidian/command.lua

```diff
@@ -216,6 +216,77 @@ command.search = function(client, data)
   end
 end
 
+--- Insert a template
+---
+---@param client obsidian.Client
+---@param data table
+command.insert_template = function(client, data)
+  if not client.opts.templates.subdir then
+    echo.err "No templates folder defined in setup()"
+    return
+  end
+
+  local templates_dir = Path:new(client.dir) / client.opts.templates.subdir
+  if not templates_dir:is_dir() then
+    echo.err(string.format("%s is not a valid directory for templates", templates_dir))
+    return
+  end
+
+  local has_telescope, _ = pcall(require, "telescope.builtin")
+  if not has_telescope then
+    echo.err "telescope.nvim is required to use the ObsidianTemplate command"
+    return
+  end
+
+  -- We need to get these upfront otherwise
+  -- Telescope hijacks the current window
+  local buf = vim.api.nvim_win_get_buf(0)
+  local win = vim.api.nvim_get_current_win()
+  local row, _ = unpack(vim.api.nvim_win_get_cursor(win))
+
+  local apply_template = function(name)
+    local template_path = Path:new(templates_dir / name)
+    local date_format = client.opts.templates.date_format or "%Y-%m-%d"
+    local time_format = client.opts.templates.time_format or "%H:%M"
+    local date = tostring(os.date(date_format))
+    local time = tostring(os.date(time_format))
+    local fp = vim.api.nvim_buf_get_name(buf)
+    local _, _, title = string.find(vim.fs.normalize(fp), ".*/(.-)%.md")
+    title = title or ""
+
+    local insert_lines = {}
+    local template_file = io.open(tostring(template_path), "r")
+    local lines = template_file:lines()
+    for line in lines do
+      line = string.gsub(line, "{{date}}", date)
+      line = string.gsub(line, "{{time}}", time)
+      line = string.gsub(line, "{{title}}", title)
+      table.insert(insert_lines, line)
+    end
+    template_file:close()
+
+    vim.api.nvim_buf_set_lines(buf, row - 1, row - 1, false, insert_lines)
+  end
+
+  local choose_template = function()
+    local opts = {
+      cwd = tostring(templates_dir),
+      attach_mappings = function(_, map)
+        map({ "i", "n" }, "<CR>", function(prompt_bufnr)
+          local template = require("telescope.actions.state").get_selected_entry()
+          apply_template(template[1])
+          require("telescope.actions").close(prompt_bufnr)
+        end)
+        return true
+      end,
+    }
+    require("telescope.builtin").find_files(opts)
+  end
+  choose_template()
+  return
```

Comment:
Could you add support for `fzf` as well if it's not too difficult? See the `ObsidianSearch` and `ObsidianQuickSwitch` commands for an example

Reply:
I have made these changes and have added support for `fzf-lua` and `fzf.vim`.

For `fzf.vim` the only way I could find to get the output from `fzf.vim` and into my lua function was to register and then shortly after remove a new command for `fzf` to send its output to (I don't know vimscript so there's probably a better way).

Finally, there is some weirdness in `fzf-lua` in which the file name it passes comes with 6 characters at the start that are not part of the filename and render as two blank spaces -- I'll open an issue with `fzf-lua`.

Reply:
Thanks @s-cassidy, I'll try it out with fzf today and see if I can figure out what's happening there